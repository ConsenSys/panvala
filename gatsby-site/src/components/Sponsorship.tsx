import React from 'react';

import WebsiteModal from './WebsiteModal';
import SponsorshipForm from './SponsorshipForm';
import Box from './system/Box';

import { withDonationFlow, IDonationFlowData } from './donationFlow';

interface Props {
  ethPrices: any;
  step: number | null;
  message: string;
  onCancel: any;
  onDonate: any;
}

const Sponsorship = (props: Props) => {
  // Click handler for donations
  async function handleDonation(values, actions) {
    console.log('Sponsorship:', 'handleDonation', values);

    const { company, firstName, lastName, email, pledgeAmount, pledgeDuration } = values;

    // Data needed for the donation flow
    const data: IDonationFlowData = {
      userData: { firstName, lastName, email, company },
      donationMetadata: {
        monthlyPledge: pledgeAmount,
        pledgeDuration,
        memo: `Sponsorship: ${company}`,
      },
      pledgeType: 'sponsorship',
      donationTier: 'Sponsor',
    };

    try {
      await props.onDonate(data, actions);
    } catch (error) {
      alert(error.message);
    }
  }

  return (
    <Box data-testid="sponsorship-container">
      <SponsorshipForm onSubmit={handleDonation} />
      <WebsiteModal
        isOpen={true}
        step={props.step}
        message={props.message}
        handleCancel={props.onCancel}
        pledgeType="sponsorship"
      />
    </Box>
  );
};

const PoweredSponsorship = withDonationFlow(Sponsorship);
export default PoweredSponsorship;
