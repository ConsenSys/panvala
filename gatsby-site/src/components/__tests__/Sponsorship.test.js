import React from 'react';
import { create } from 'react-test-renderer';
import Sponsorship from '../Sponsorship';

describe('Donation', () => {
  let component, instance;

  beforeEach(() => {
    component = create(<Sponsorship />);
    instance = component.getInstance();
  });

  test('should render component and initialize with correct state', () => {
    const initialState = {
      email: '',
      fullName: '',
      selectedAccount: '',
      step: null,
      error: false,
      message: '',
      tier: '',
      panPurchased: 0,
    };

    expect(component.toJSON()).toMatchSnapshot();
    expect(instance.state).toEqual(initialState);
  });
});
