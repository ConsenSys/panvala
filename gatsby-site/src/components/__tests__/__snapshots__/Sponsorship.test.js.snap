// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Donation should render component and initialize with correct state 1`] = `
<form
  className="w-80-l w-90-m w-100 center"
  name="sponsorship-pledge"
  onSubmit={[Function]}
>
  <div
    className="tl mt4"
  >
    <label
      className="ma0 f6 mb3 black-40"
      required={true}
    >
      Company
      <b
        className="red f7"
      >
         *
      </b>
    </label>
  </div>
  <div
    className="red tl pt2"
  />
  <input
    className="sc-bdVaJa kGEDHa input-reset b--black-10"
    id="pledge-company"
    label="Company"
    name="company"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Enter the name of your company"
    required={true}
    type="text"
    value=""
  />
  <div
    className="tl mt4"
  >
    <label
      className="ma0 f6 mb3 black-40"
    >
      First Name
    </label>
  </div>
  <div
    className="red tl pt2"
  />
  <input
    className="sc-bdVaJa kGEDHa input-reset b--black-10"
    id="pledge-first-name"
    label="First Name"
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Enter your first name"
    type="text"
    value=""
  />
  <div
    className="tl mt4"
  >
    <label
      className="ma0 f6 mb3 black-40"
    >
      Last Name
    </label>
  </div>
  <div
    className="red tl pt2"
  />
  <input
    className="sc-bdVaJa kGEDHa input-reset b--black-10"
    id="pledge-last-name"
    label="Last Name"
    name="lastName"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Enter your last name"
    type="text"
    value=""
  />
  <div
    className="tl mt4"
  >
    <label
      className="ma0 f6 mb3 black-40"
      required={true}
    >
      Email
      <b
        className="red f7"
      >
         *
      </b>
    </label>
  </div>
  <div
    className="red tl pt2"
  />
  <input
    className="sc-bdVaJa kGEDHa input-reset b--black-10"
    id="pledge-email"
    label="Email"
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Enter your email address"
    required={true}
    type="text"
    value=""
  />
  <div
    className="tl mt4"
  >
    <label
      className="ma0 f6 mb3 black-40"
      required={true}
    >
      Monthly Pledge Amount (USD)
      <b
        className="red f7"
      >
         *
      </b>
    </label>
  </div>
  <div
    className="red tl pt2"
  />
  <input
    className="sc-bdVaJa kGEDHa input-reset b--black-10"
    id="pledge-amount"
    label="Monthly Pledge Amount (USD)"
    min="0"
    name="pledgeAmount"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Enter your pledge amount"
    required={true}
    step="1"
    type="number"
    value=""
  />
  <div
    className="tl mt4"
  >
    <label
      className="ma0 f6 mb3 black-40"
      required={true}
    >
      How many months of your pledge will you prepay today?
      <b
        className="red f7"
      >
         *
      </b>
    </label>
  </div>
  <div
    className="red tl pt2"
  />
  <select
    className="f6 input-reset b--black-10 pv3 ph2 db center w-100 br3 mt2 bg-white black-50"
    id="pledge-duration-select"
    name="pledgeDuration"
    onBlur={[Function]}
    onChange={[Function]}
    required={true}
    value={0}
  >
    <option
      defaultValue="0"
      disabled=""
      value="0"
    >
      Select the amount of months you would like to prepay for
    </option>
    <option
      value="1"
    >
      1 month
    </option>
    <option
      value="3"
    >
      3 months
    </option>
    <option
      value="6"
    >
      6 months
    </option>
    <option
      value="12"
    >
      12 months
    </option>
  </select>
  <img
    alt=""
    className="fr mr2 o-50"
    src="test-file-stub"
    style={
      Object {
        "marginTop": "-35px",
      }
    }
  />
  <div>
    <input
      className="f6 link dim bn br-pill pv3 ph4 white bg-teal fw7 mt4"
      disabled={false}
      name="submit"
      type="submit"
      value="Sponsor"
    />
  </div>
</form>
`;
